internal/controller/system/controller.go

```diff
@@ -127,7 +130,16 @@ func (ctrl *DefaultController) CreateLedger(ctx context.Context, name string, co
 
 func (ctrl *DefaultController) GetLedger(ctx context.Context, name string) (*ledger.Ledger, error) {
 	return tracing.Trace(ctx, ctrl.tracerProvider.Tracer("system"), "GetLedger", func(ctx context.Context) (*ledger.Ledger, error) {
-		return ctrl.store.GetLedger(ctx, name)
+		l, err := ctrl.store.GetLedger(ctx, name)
+		if err != nil {
+			return nil, err
+		}
+		
+		if l.DeletedAt != nil {
```

Comment:
Is the ledger supposed to have a DeletedAt? Or is this the ledger's bucket DeletedAt? If so, is that what we want?